### PR TITLE
Apply valid name functionality 

### PR DESCRIPTION
**Pull request by suaas21**
fixes: #924

### CODE REVIEW

File: pkg/controller/restore_session.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ import (
 )
 
 const (
-	RestoreJobPrefix                = "stash-restore-"
+	RestoreJobPrefix                = "restore-"
```
Comment (tamalsaha) [inline pkg/controller/restore_session.go]: `-` should be removed?

File: pkg/controller/backup_session.go [hunk 2]
```diff
@@ -54,8 +54,8 @@ import (
 )
 
 const (
-	BackupJobPrefix                = "stash-backup-"
-	VolumeSnapshotPrefix           = "volume-snapshot-"
+	BackupJobPrefix                = "backup-"
```
Comment (tamalsaha) [inline pkg/controller/backup_session.go]: `-` should be removed, since `VaildName***` methods add - .

File: pkg/util/util.go [hunk 3]
```diff
@@ -472,3 +472,31 @@ func UpsertInterimVolume(kubeClient kubernetes.Interface, podSpec core.PodSpec,
...
+	case apis.KindStatefulSet:
+		return "sts"
+	case apis.KindPersistentVolumeClaim:
+		return "pvc"
+	case apis.KindPod:
+		return "po"
+	case apis.KindAppBinding:
+		return "ab"
+	case api_v1beta1.ResourceKindBackupConfiguration:
+		return "bc"
```
Comment (tamalsaha) [inline pkg/util/util.go]: `api_v1beta1` types should use our constants from api package.

File: pkg/util/util.go [hunk 4]
```diff
@@ -472,3 +472,31 @@ func UpsertInterimVolume(kubeClient kubernetes.Interface, podSpec core.PodSpec,
 	}
 	return AttachPVC(podSpec, volumes, volumeMounts), nil
 }
+
+func ResourceKindShortForm(kind string) string {
+	switch kind {
+	case apis.KindDeployment:
```
Comment (tamalsaha) [inline pkg/util/util.go]: For official types match the short names defined here: https://kubernetes.io/docs/reference/kubectl/overview/#resource-types

File: pkg/util/util.go [hunk 5]
```diff
@@ -472,3 +472,31 @@ func UpsertInterimVolume(kubeClient kubernetes.Interface, podSpec core.PodSpec,
...
+		return "deploy"
+	case apis.KindReplicationController:
+		return "rc"
+	case apis.KindDaemonSet:
+		return "dmn"
```
Comment (tamalsaha) [inline pkg/util/util.go]: `"ds"`

File: pkg/util/util.go [hunk 6]
```diff
@@ -472,3 +472,31 @@ func UpsertInterimVolume(kubeClient kubernetes.Interface, podSpec core.PodSpec,
...
+	case apis.KindStatefulSet:
+		return "sts"
+	case apis.KindPersistentVolumeClaim:
+		return "pvc"
+	case apis.KindPod:
+		return "po"
+	case apis.KindAppBinding:
+		return "ab"
```
Comment (tamalsaha) [inline pkg/util/util.go]: Where is this prefix used? Probably should be `app` .

File: pkg/util/util.go [hunk 7]
```diff
@@ -23,7 +23,7 @@ import (
 	"strings"
 
 	"stash.appscode.dev/stash/apis"
-	api_v1beta1 "stash.appscode.dev/stash/apis/stash/v1beta1"
+	apis_v1beta1 "stash.appscode.dev/stash/apis/stash/v1beta1"
```
Comment (tamalsaha) [inline pkg/util/util.go]: `apis_v1beta1` -> `api_v1beta1`

File: apis/constants.go [hunk 8]
```diff
@@ -76,4 +76,13 @@ const (
 	StashInterimVolume          = "stash-interim-volume"
 	StashInterimVolumeMountPath = "/stash-interim-volume"
 	StashInterimDataDir         = "/stash-interim-volume/data"
+
+	KindBackupConfiguration = "BackupConfiguration"
```
Comment (tamalsaha) [inline apis/constants.go]: Remove these and use the `ResourceKind***` constants.

File: pkg/util/util.go [hunk 9]
```diff
@@ -472,3 +472,33 @@ func UpsertInterimVolume(kubeClient kubernetes.Interface, podSpec core.PodSpec,
...
+	case apis.KindBackupSession:
+		return "bs"
+	case apis.KindBackupBlueprint:
+		return "bb"
+	case apis.KindRestoreSession:
+		return "rs"
+	case apis.KindRepository:
+		return "repo"
+	}
+	return strings.ToLower(kind)
```
Comment (tamalsaha) [inline pkg/util/util.go]: Make it `default:`

File: pkg/util/util.go [hunk 10]
```diff
@@ -472,3 +472,33 @@ func UpsertInterimVolume(kubeClient kubernetes.Interface, podSpec core.PodSpec,
...
+		return "ds"
+	case apis.KindStatefulSet:
+		return "sts"
+	case apis.KindPersistentVolumeClaim:
+		return "pvc"
+	case apis.KindPod:
+		return "po"
+	case apis.KindAppBinding:
+		return "app"
+	case apis.KindBackupConfiguration:
```
Comment (tamalsaha) [inline pkg/util/util.go]: Remove unused constants.

File: pkg/util/util.go [hunk 11]
```diff
@@ -472,3 +472,24 @@ func UpsertInterimVolume(kubeClient kubernetes.Interface, podSpec core.PodSpec,
 	}
 	return AttachPVC(podSpec, volumes, volumeMounts), nil
 }
+
```
Comment (suaas21) [inline pkg/util/util.go]: Add reff comment

**Pull request merged**